akka-projection-core/src/main/scala/akka/projection/internal/OffsetSerialization.scala

```diff
@@ -51,11 +49,11 @@ import akka.serialization.Serializers
     val offset: Offset = rep match {
       case SingleOffset(_, manifest, offsetStr, _) => fromStorageRepresentation[Offset](offsetStr, manifest)
       case MultipleOffsets(reps) =>
-        val offsets: Map[StringKey, Inner] = reps.map {
+        val offsets: Map[String, Inner] = reps.map {
```

Comment:
This is another point where we could get a ClassCastException. We can have a MergeableKey with StringKey impl and other user defined keys. The OffsetSerialization only know how to deal with one type. So, we must go back to plain string. At the end, we are always using plain string anyway.